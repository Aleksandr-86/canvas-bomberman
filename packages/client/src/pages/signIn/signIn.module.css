.signIn {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 350px;
  padding: 10px 0;
  margin: 30px 0;
}

.signInHeader {
  font-size: 40px;
}

.signInContent {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
  margin-top: 100px;
}

.signInAction {
  display: flex;
  flex-direction: inherit;
  align-items: center;
  width: 100%;
  margin-top: 100px;
}

.actionButton {
  margin-top: 30px;
}
