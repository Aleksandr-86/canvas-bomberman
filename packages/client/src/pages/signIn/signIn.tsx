import React, { useEffect, useState } from 'react'

import { Link } from 'react-router-dom'
import { Button } from '../../components/button/button'
import { FormField } from '../../components/formField/formField'

import { useForm, Validations } from '../../hooks/useForm'
import { getValidations } from '../../features/validation'

import styles from './signIn.module.css'
import baseStyles from '../../app/app.module.css'

type FormType = {
  displayName: string
  password: string
}

export const SignIn: React.FC = () => {
  const [validations, setValidations] = useState<Validations<FormType>>()

  useEffect(() => {
    setValidations(getValidations(['displayName', 'password']))
  }, [])

  const { values, errors, register, watch, isValid } = useForm<FormType>({
    validations: validations,
  })

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
  }

  return (
    <>
      <form className={styles.signIn} onSubmit={handleSubmit}>
        <div className={styles.signInHeader}>Вход</div>

        <div className={styles.signInContent}>
          <FormField
            label="Логин"
            name="name"
            type="text"
            {...register('displayName')}></FormField>
          <FormField
            label="Пароль"
            name="password"
            type="password"
            {...register('password')}></FormField>
        </div>

        <div className={styles.signInAction}>
          <div className={styles.actionButton}>
            <Button type="submit">Все верно!</Button>
          </div>

          <div className={styles.actionButton}>
            <Link className={baseStyles.linkButton} to="/">
              На Главную
            </Link>
          </div>
        </div>
      </form>
    </>
  )
}
