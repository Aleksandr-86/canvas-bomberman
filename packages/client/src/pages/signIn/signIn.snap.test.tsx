import * as renderer from 'react-test-renderer'
import { SignIn } from './signIn'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore([])
const store = mockStore({})

describe(`<SignIn/> snapshot test cases`, () => {
  test(`<SignIn/> should render <form className=signIn> parent with child elements`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/signIn`]}>
            <SignIn />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
