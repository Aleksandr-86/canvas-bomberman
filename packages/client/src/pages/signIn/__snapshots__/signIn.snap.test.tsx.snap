// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn/> snapshot test cases <SignIn/> should render <form className=signIn> parent with child elements 1`] = `
<form
  className="signIn"
  onSubmit={[Function]}
>
  <div
    className="signInHeader"
  >
    Вход
  </div>
  <div
    className="signInContent"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Логин
      </label>
      <div>
        <input
          className="input"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <p
        className="error"
      />
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Пароль
      </label>
      <div>
        <input
          className="input"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <p
        className="error"
      />
    </div>
  </div>
  <div
    className="signInAction"
  >
    <div
      className="actionButton"
    >
      <button
        className="button"
        type="submit"
      >
        Всё верно!
      </button>
    </div>
    <div
      className="actionButton"
    >
      <a
        className="linkButton"
        href="/"
        onClick={[Function]}
      >
        На Главную
      </a>
    </div>
  </div>
</form>
`;
